pkg/api/agentPoolOnlyApi/vlabs/types.go

```diff
@@ -30,15 +30,37 @@ type ManagedCluster struct {
 // Properties represents the ACS cluster definition
 type Properties struct {
 	ProvisioningState       ProvisioningState        `json:"provisioningState,omitempty"`
-	KubernetesVersion       string                   `json:"kubernetesVersion"`
+	KubernetesVersion       string                   `json:"kubernetesVersion,omitempty"`
+	OrchestratorProfile     *OrchestratorProfile     `json:"orchestratorProfile,omitempty"`
 	DNSPrefix               string                   `json:"dnsPrefix" validate:"required"`
 	FQDN                    string                   `json:"fqdn,omitempty"`
 	AgentPoolProfiles       []*AgentPoolProfile      `json:"agentPoolProfiles,omitempty" validate:"dive,required"`
 	LinuxProfile            *LinuxProfile            `json:"linuxProfile,omitempty" validate:"required"`
 	WindowsProfile          *WindowsProfile          `json:"windowsProfile,omitempty"`
 	ServicePrincipalProfile *ServicePrincipalProfile `json:"servicePrincipalProfile,omitempty"`
 	AccessProfiles          map[string]AccessProfile `json:"accessProfiles,omitempty"`
-	CertificateProfile      *CertificateProfile      `json:"certificateProfile,omitempty" validate:"required"`
+	CertificateProfile      *CertificateProfile      `json:"certificateProfile,omitempty"`
+	AzProfile               *AzProfile               `json:"azProfile,omitempty"`
```

Comment:
@jim-minter @pweil- fyi, we are not getting rid of azProfile with MSI - MSI is one more field in azure.conf